controllers/image_generator.go

```diff
@@ -644,6 +651,17 @@ func (r *ImageGenerator) createImageConfigMapFromFile() error {
 		return err
 	}
 
+	// For Azure provider set the IMAGE_GALLERY_NAME if its empty
+	// The IMAGE_GALLERY_NAME is set to IMAGE_GALLERY_NAME_PREFIX + "_" + clusterID
+	if r.provider == AzureProvider {
+		if cm.Data["IMAGE_GALLERY_NAME"] == "" {
+			image_gallery_name := cm.Data["IMAGE_GALLERY_NAME_PREFIX"] + "_" + r.clusterId
+			cm.Data["IMAGE_GALLERY_NAME"] = image_gallery_name
```

Comment:
Should we check that the generated name is valid ?

```
  # Note that the gallery name must be unique across the subscription and not exceed 80 characters.
  # Also, the allowed characters are English alphanumeric characters,
  # with underscores and periods allowed in the middle
```

Reply:
Based on this comment https://github.com/openshift/sandboxed-containers-operator/pull/404#discussion_r1618405423
if I change the code to 
```
const osc_gallery_name_prefix = PodVMGallery
image_gallery_name :=  osc_gallery_name_prefix + "_" + r.clusterId
```
This will always be valid as clusterId is only the first 8 chars. So imho no need to recheck if the generated name is valid. Wdyt ?